File: BackEnd/WeShare.Application/Dtos/Transaction/UpdateTransactionDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WeShare.Core.Enums;

namespace WeShare.Application.Dtos.Transaction
{
    public class UpdateTransactionDto
    {
        public int Id { get; set; }
        public string Description { get; set; }
        public decimal? Amount { get; set; }
        public Dictionary<int, decimal>? SplitAmounts { get; set; }
        public SplitStrategyEnum Strategy { get; set; }
        public List<int> DebtIds { get; set; }
    }
}

File: FrontEnd/src/types/transaction/UpdateTransactionDto.ts
import type { SplitStrategyEnum } from "@/constants/SplitStrategyEnum"

export interface UpdateTransactionDto {
  id: number
  description: string
  amount?: number
  splitAmounts?: Record<number, number> | null
  strategy: SplitStrategyEnum
  debtIds: number[]
}
